substrate/frame/support/procedural/src/pallet/parse/config.rs

```diff
@@ -352,8 +336,23 @@ impl ConfigDef {
 		};
 
 		let has_frame_system_supertrait = item.supertraits.iter().any(|s| {
-			syn::parse2::<ConfigBoundParse>(s.to_token_stream())
-				.map_or(false, |b| b.0 == *frame_system)
+			syn::parse2::<syn::Path>(s.to_token_stream()).map_or(false, |b| {
+				let mut expected_system_config = frame_system.clone();
+				expected_system_config
+					.segments
+					.push(syn::PathSegment::from(syn::Ident::new("Config", b.span())));
+				// the parse path might be something like `frame_system::Config<...>`, so we
```

Comment:
same, what I am doing here is somewhat new and I didn't originally add tests for it.